core/src/main/java/org/jruby/RubyBoolean.java

```diff
@@ -192,6 +205,19 @@ public static IRubyObject true_xor(IRubyObject t, IRubyObject oth) {
         public static RubyString true_to_s(IRubyObject t) {
             return RubyString.newStringShared(t.getRuntime(), TRUE_BYTES);
         }
+
+        /*
+        Because Boolean objects can't cast to boolean via Class.cast, and you can't return
+        a primitive from a generified method, this impl remains ungenerified.
+         */
+        @Override
+        public Object toJava(Class target) {
+            if (target.isAssignableFrom(Boolean.class) | target.equals(boolean.class)) {
```

Comment:
same here `||` got lost as `|`

Reply:
Ok it turns out isAssignableFrom is required here because the contract of Ruby Boolean is that it will coerce to a Boolean object for java.lang.Boolean or any supertype up to java.lang.Object.

I'll make the other changes along with my fixes.